src/Tasks/AssemblyDependency/ReferenceTable.cs

```diff
@@ -1781,31 +1782,26 @@ private bool FindAssociatedFiles()
                 }
             }
 
-            // Add each new dependency found.
-            foreach (KeyValuePair<AssemblyNameExtension, Reference> newEntry in newEntries)
-            {
-                newDependencies = true;
-                AddReference(newEntry.Key, newEntry.Value);
-            }
-
-            return newDependencies;
+            return false;
         }
 
         /// <summary>
         /// Resolve all references that have not been resolved yet to real files on disk.
         /// </summary>
-        private void ResolveAssemblyFilenames()
+        private int ResolveAssemblyFilenames()
         {
-            foreach (AssemblyNameExtension assemblyName in References.Keys)
+            int referencesResolved = 0;
+            foreach (KeyValuePair<AssemblyNameExtension, Reference> kvp in References)
             {
-                Reference reference = GetReference(assemblyName);
-
+                Reference reference = kvp.Value;
                 // Has this reference been resolved to a file name?
                 if (!reference.IsResolved && !reference.IsUnresolvable)
                 {
-                    ResolveReference(assemblyName, null, reference);
+                    ResolveReference(kvp.Key, null, reference);
```

Comment:
If the reference couldn't be resolved would we get an exception thrown here?

Reply:
I think (with low confidence) that ResolveReference avoids throwing exceptions, instead appending to a list of exceptions. @rainersigwald? In any case, although you're right that that would mess with the references count, I don't think we care about measuring perf for projects that throw exceptions.

Reply:
If that's the case, is there a way we can check if that exceptions list has been added to, where only then should we increment?